osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CreateServiceFunctionChainTaskTest.java

```diff
@@ -0,0 +1,171 @@
+import static org.osc.core.broker.service.tasks.conformance.openstack.sfc.CreateServiceFunctionChainTaskTestData.*;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityTransaction;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osc.core.broker.model.entities.virtualization.SecurityGroup;
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.model.sdn.NetworkElementImpl;
+import org.osc.core.broker.service.exceptions.VmidcException;
+import org.osc.core.broker.service.persistence.SecurityGroupEntityMgr;
+import org.osc.core.broker.service.test.InMemDB;
+import org.osc.core.broker.util.TransactionalBroadcastUtil;
+import org.osc.core.broker.util.db.DBConnectionManager;
+import org.osc.core.test.util.TestTransactionControl;
+import org.osc.sdk.controller.api.SdnRedirectionApi;
+import org.osc.sdk.controller.element.NetworkElement;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreateServiceFunctionChainTaskTest {
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+	@Mock
+	private EntityManager em;
+
+	@Mock
+	private EntityTransaction tx;
+
+	@Mock(answer = Answers.CALLS_REAL_METHODS)
+	private TestTransactionControl txControl;
+
+	@Mock
+	private DBConnectionManager dbMgr;
+
+	@Mock
+	private TransactionalBroadcastUtil txBroadcastUtil;
+
+	@Mock
+	private ApiFactoryService apiFactoryServiceMock;
+
+	@Mock
+	private SdnRedirectionApi sdnApi;
+
+	@InjectMocks
+	private CreateServiceFunctionChainTask task;
+
+	@Before
+	public void testInitialize() throws VmidcException {
+		MockitoAnnotations.initMocks(this);
+		this.em = InMemDB.getEntityManagerFactory().createEntityManager();
+
+		this.txControl.setEntityManager(this.em);
+		Mockito.when(this.dbMgr.getTransactionalEntityManager()).thenReturn(this.em);
+		Mockito.when(this.dbMgr.getTransactionControl()).thenReturn(this.txControl);
+
+		populateDatabase();
+	}
+
+	@AfterClass
+	public static void testTearDowm() {
+		InMemDB.shutdown();
+	}
+
+	@Test
+	public void testExecute_WithSecurityGroupSFCBinded_ExpectCreate() throws Exception {
+		// Arrange.
+		CreateServiceFunctionChainTask task = this.task.create(
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getServiceFunctionChain(),
+				SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE);
+		registerNetworkElement(SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE, new NetworkElementImpl("NETWORK_ELEMENT_ID_2"));
+
+		// Act.
+		String taskName = task.getName();
+
+		// Assert
+		assertEquals(SecurityGroupEntityMgr.findById(this.em, SECURITY_GROUP_SFC_BINDED_EXPECT_CREATE.getId())
+				.getNetworkElementId(), "NETWORK_ELEMENT_ID_2");
```

Comment:
Where is the "NETWORK_ELEMENT_ID_2" string from? If defined in two places use a shared field.

Reply:
@larkinscarvalho I think you missed this comment.